src/frontend/OverlayManager/Overlay/NavBarOverlayWrapper/index.tsx

```diff
@@ -3,13 +3,51 @@ import styles from './index.module.scss'
 import { t } from 'i18next'
 import { NavBarOverlay, NavItem, Images, SubLink } from '@hyperplay/ui'
 import { Link, useLocation } from 'react-router-dom'
+import { BrowserGameProps } from 'frontend/OverlayManager/types'
+import { useQuery } from '@tanstack/react-query'
+import classNames from 'classnames'
 
-export function NavBarOverlayWrapper() {
+export function NavBarOverlayWrapper({ appName, runner }: BrowserGameProps) {
   const location = useLocation()
-  const { pathname } = location
+  const { pathname, search } = location
+  const searchParams = new URLSearchParams(search)
+  const urlQueryParam = searchParams.get('url')
   const [collapsed, setCollapsed] = useState(false)
   const comingSoonText = t('overlay.comingSoon', 'Coming Soon')
   const [mmCollapsed, mmSetCollapsed] = useState(false)
+  const [mktCollapsed, setMktCollapsed] = useState(false)
+
+  const { data } = useQuery({
+    queryKey: ['getGameInfo', appName],
+    queryFn: async () => {
+      return window.api.getGameInfo(appName, runner)
+    }
+  })
+
+  const marketplaceEnabled =
+    !!data && !!data.networks?.some((val) => !!val.marketplace_urls?.length)
+  const marketplaceClass: Record<string, boolean> = {}
+  marketplaceClass[styles.disabled] = !marketplaceEnabled
+
+  const mktplaceSublinks = data?.networks
+    ?.filter((val) => !!val.marketplace_urls)
+    .flatMap((val) =>
+      val
+        .marketplace_urls!.filter((val) => !!val)
```

Comment:
if marketplace_urls can be null default to an empty array

```ts
.(marketplace_urls ?? []).filter((val) => !!val)
```

Reply:
It can't be null since I filter prior with `!!val.marketplace_urls` but I can add the `?? []` to get rid of the `!` which is good in case the above line is changed or removed in the future